### ISSUE TITLE
`used-before-assignment` false positive with walrus operator

### ISSUE THREAD

**Issue by DetachHead** [PyCQA/pylint#4828]
### Bug description

`used-before-assignment` false positive occurs on the following code:
```python
def foo() -> str:
    return bar if (bar := "") else ""
```

### Command used

```shell
pylint foo.py
```

### Pylint output

```shell
E0601: Using variable 'bar' before assignment (used-before-assignment)
```

### Expected behavior

no error

### Pylint version

```shell
pylint 2.9.6
astroid 2.6.6
Python 3.10.0b4 (v3.10.0b4:2ba4b20854, Jul  9 2021, 21:56:21) [Clang 12.0.5 (clang-1205.0.22.9)]
```

**Comment by Pierre-Sassoulas** [comment]
Thank you for opening the issue I can reproduce this.

**Comment by Pierre-Sassoulas** [comment]
@nyuszika7h could you open another issue, please ? It look like it's affecting another node than a return in your case.
